config/manifests/kustomization.yaml

```diff
@@ -3,6 +3,7 @@
 resources:
 - ../default
 - ../scorecard
+- codeflare-operator.clusterserviceversion.yaml
```

Comment:
We have to find a way to keep the file, but not have it in the manifests overlay, otherwise it'll be taken into account by the ODH operator.

Reply:
I don't think there is a way to run the `make bundle` command without the CSV in the kustomization file which is what causes the spec.Icon error in the previous commit OLM workflow.

Reply:
Right, now that the `manifests` directory is used by the ODH operator, we need to move the CSV file into a separate directory / overlay. WDYT?

Reply:
I can test this out :+1:

Reply:
It seems no matter where we place the CSV file we will still get the 
```Error: Value : (codeflare-operator.v0.0.0-dev) csv.Spec.Icon elements should contain both data and mediatype``` error when running `make bundle`.

Reply:
Hum, that sounds like the CSV hasn't been taken into account. The Icon field is in there no?

Anyhow, the CSV cannot be included into the set of manifests that the ODH operator manages for the codeflare component.

Reply:
Thanks, that confirms the CSV wasn't properly taken into account while moved to another directory. We should look into it, so it can be moved out the `manifests` directory, without breaking the OLM bundle generation.

Reply:
I read up on the `operator-sdk generate bundle` command and I found this.
>A CSV manifest is generated by collecting data from the set of manifests passed to this command (see below),
such as CRDs, RBAC, etc., and applying that data to a "base" CSV manifest. This base CSV can contain metadata,
added by hand or by the 'generate kustomize manifests' command, and can be passed in like any other manifest
(see below) or by file at the exact path '<kustomize-dir>/bases/<package-name>.clusterserviceversion.yaml'.
Be aware that 'generate bundle' idempotently regenerates a bundle, so all non-metadata values in a base
will be overwritten. If no base was passed in, input manifest data will be applied to an empty CSV.

I assume the base in this case would be the CSV file we would normally pass into the manifests folder. So the input data is applied to an empty CSV in our case because we are not supplying. 

I inspected the CSV generated in the bundle folder and the icon and media type are empty strings.

The bundle generates just fine it is just the validation that is an issue.
It seems we can skip certain validation as seen [here](https://github.com/operator-framework/operator-sdk/issues/4375#issuecomment-756896868) 

Doing some more reading on this and I can see there is this documentation on it:
```
To validate a bundle against the validator for operatorhub.io specifically, in addition to required bundle validators:

  $ operator-sdk bundle validate ./bundle --select-optional name=operatorhub
```
If we opted to use this method of validation we would miss out on validating against the operator framework.
What are your thoughts @astefanutti @sutaakar

Reply:
@astefanutti What about keeping CSV in manifests directory and moving ODH related manifests to some other directory?

Reply:
@sutaakar that could be a solution, though that change will have to be coordinated with the ODH operator so it sources the new directory. Actually, it'd be less confusing to have an `odh` or `opendatahub` directory to host these resources, rather than inside the existing `manifests` directory. We need to check with @zdtsw if that's an option.

Reply:
if i understand it correct, what you try to achieve is to do the logic change in your upstream "codeflare-project" and sync down to poendatahub's codeflare-operator with auto update this line  `odh-codeflare-operator-controller-image=quay.io/project-codeflare/codeflare-operator:$somenew_version` ?  then what has it to do with CSV file : thinking : in ODH


in ODH, operator goes to the folder you defined (in current code base is your `config/manifests`) and apply result of  `kustomize build`. I do not think it will matter if you want to restructure your config as long as ODH operator has the correct path with all the resourced defined within.

Reply:
@zdtsw your understanding is correct. The CSV resource must not be part of the directory sourced by ODH. The issue being that operator-sdk is opinionated with using the `manifests` directory, which is currently used by ODH. We would need to change the path for the CodeFlare operator in ODH to something different. Would you have any objections?

Reply:
no objection from my side. 
all we need is when you are ready to have such change in ODH codeflare, let us (operator) know so we can do any modification if needed.

Reply:
Sounds good, thanks @zdtsw!

Reply:
So I'll put the CSV file back into manifests and should we remove the params files and place them into a new directory for just the ODH CFO?

Reply:
@Bobbins228 yes, and make sure the `odh-codeflare-operator-controller-image` variable is correctly taken into account in the Deployment.

Reply:
Okay. `manager.yaml` is seemingly the only file that takes in the `IMG` variable into account so what I can do is replace the [image replacement](https://github.com/project-codeflare/codeflare-operator/blob/main/Makefile#L287) with a command that will replace the image in params.env with the `IMG` variable.

I can leave the param files in manager.yaml and it will be updated with the new image.